#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *;
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *;

from math import pi

gen = ParameterGenerator ()
gen.add("max_iteration", int_t, 0, "the maximum iteration", 100, 0, 1000)
gen.add("correspondence_distance", double_t, 0,
                                   "distance to ignore to calculate correspondence",
                                   10, 0, 100)
gen.add("transform_epsilon", double_t, 0, "transform epsion to stop iteration", 1e-9, 1e-9, 0.1)
gen.add("euclidean_fittness_epsilon", double_t, 0, "euclidean fittness epsilon", 0.01, 0.0001, 1.0)
exit (gen.generate (PACKAGE, "jsk_pcl_ros", "ICPRegistration"))
